temporal/fx.go

```diff
@@ -593,25 +593,12 @@ func ApplyClusterMetadataConfigProvider(
 			return config.ClusterMetadata, config.Persistence, fmt.Errorf("error while fetching cluster metadata: %w", err)
 		}
 
-		// Verify current cluster metadata
-		var persistedShardCount = resp.HistoryShardCount
-		if config.Persistence.NumHistoryShards != persistedShardCount {
-			logger.Warn(
-				mismatchLogMessage,
-				tag.Key("persistence.numHistoryShards"),
-				tag.IgnoredValue(config.Persistence.NumHistoryShards),
-				tag.Value(persistedShardCount))
-			config.Persistence.NumHistoryShards = persistedShardCount
-		}
-		if resp.FailoverVersionIncrement != clusterData.FailoverVersionIncrement {
-			logger.Warn(
-				mismatchLogMessage,
-				tag.Key("clusterMetadata.FailoverVersionIncrement"),
-				tag.IgnoredValue(clusterData.FailoverVersionIncrement),
-				tag.Value(resp.FailoverVersionIncrement))
-			config.ClusterMetadata.FailoverVersionIncrement = resp.FailoverVersionIncrement
-		}
 		if resp.IsGlobalNamespaceEnabled != clusterData.EnableGlobalNamespace {
```

Comment:
I'm not sure it should be allowed to downgrade for IsGlobalNamespaceEnabled from true to false. What is the implication?

Reply:
Updated